lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskStatus.cs

```diff
@@ -246,12 +252,20 @@ public TaskStatusProto ToProto()
                 }
                 else if (_lastException.IsPresent())
                 {
+                    byte[] error;
+                    try
+                    {
+                        error = SerializeException(_lastException.Value);
+                    }
+                    catch (SerializationException)
+                    {
+                        error = SerializeException(new NonSerializableTaskException(_lastException.Value.ToString()));
```

Comment:
What does "cause" refer to? `Exception.ToString()` provides both the StackTrace and the Exception message.